include/deal.II/base/parallel.h

```diff
@@ -542,20 +542,37 @@ namespace parallel
                           const Function                             &f,
                           const unsigned int                          grainsize)
   {
-#ifndef DEAL_II_WITH_TBB
-    // make sure we don't get compiler
-    // warnings about unused arguments
-    (void)grainsize;
+#ifdef DEAL_II_WITH_TASKFLOW
+    tf::Executor &executor = MultithreadInfo::get_taskflow_executor();
+    tf::Taskflow  taskflow;
+
+    const std::size_t           n = end - begin;
+    tf::IndexRange<std::size_t> range(0, n, 1);
+
+    taskflow.for_each_by_index(
+      range,
+      [&](tf::IndexRange<std::size_t> subrange) {
+        auto subrange_begin = begin + subrange.begin();
+        auto subrange_end   = subrange_begin + subrange.size();
```

Comment:
Not every iterator will have `operator+(int)`. I think you want to write this as 
```suggestion
        auto subrange_begin = begin;
        std::advance(subrange_begin,  subrange.begin());
        auto subrange_end   = subrange_begin;
        std::advance(subrange_end, subrange.size());
```